app/Http/Controllers/SearchController.php

```diff
@@ -15,10 +15,8 @@ public function search(Request $request)
         $q = $request->input('q');
 
 
-        $book = Books::where('name', 'LIKE', '%' . $q . '%')
-                    ->get();
-        $author = Authors::where('name', 'LIKE', '%' . $q . '%')
-                    ->get();
+        $book = Books::where('name', 'LIKE', '%' . $q . '%')->get();
```

Comment:
try something like this

$book = Books::where()->select([col1, col2, DB::raw('"book" as type')]);
$author = Author::where()->select(/* identical as above, with different type*/)->union($book);

$autor->paginate();

Should give you a nice result where you no longer need to do if statements and count on objects (array of objects)

Small suggestion